incubator/hnc/third_party/open-policy-agent/gatekeeper/pkg/webhook/certs.go

```diff
@@ -580,18 +570,45 @@ func (r *ReconcileWH) Reconcile(request reconcile.Request) (reconcile.Result, er
 	return reconcile.Result{}, nil
 }
 
-func (r *ReconcileWH) ensureCRDConvWHCerts(certPem []byte) (reconcile.Result, error) {
+func (r *ReconcileWH) ensureVWHCerts(certPem []byte) error {
+	vwh := &unstructured.Unstructured{}
+	vwh.SetGroupVersionKind(vwhGVK)
+	if err := r.client.Get(r.ctx, r.vwhKey, vwh); err != nil {
+		if k8sErrors.IsNotFound(err) {
+			crLog.Info("VWK " + r.vwhKey.Name + " is not found. No action is needed.")
+			return nil
+		}
+		// Error reading the object - requeue the request.
+		return err
+	}
+
+	crLog.Info("ensuring CA cert on ValidatingWebhookConfiguration")
+	if err := injectCertToWebhook(vwh, certPem); err != nil {
+		crLog.Error(err, "unable to inject cert to webhook")
+		return err
+	}
+	if err := r.client.Update(r.ctx, vwh); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (r *ReconcileWH) ensureCRDConvWHCerts(certPem []byte) error {
```

Comment:
Document that this function essentially returns an arbitrary error if multiple errors are encountered; if it encounters multiple errors, these are logged but not returned.